scripts/docker-compose.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+set -e
+
+# Function to add or update a key-value pair in the .env file
+update_env() {
+  key=$1
+  value=$2
+  if grep -q "^${key}=" "$ENV_FILE"; then
+    # If key exists, update it
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+      sed -i '' "s|^${key}=.*|${key}=${value}|" "$ENV_FILE"
+    else
+      sed -i "s|^${key}=.*|${key}=${value}|" "$ENV_FILE"
+    fi
+    echo "Updated ${key} in .env file"
+  else
+    echo "${key}=${value}" >>"$ENV_FILE"
+    echo "Added ${key} to .env file"
+  fi
+}
+
+ROOT_DIR="$(dirname "$(dirname "$(readlink -f "$0")")")"
+ENV_FILE="${ROOT_DIR}/.env"
+ENV_SAMPLE="${ROOT_DIR}/.env.sample"
+
+# Check if .env exists, if not copy .env.sample
+if [ ! -f "$ENV_FILE" ]; then
+  cp "$ENV_SAMPLE" "$ENV_FILE"
+  echo ".env file created from .env.sample"
+  cat <<EOF >>"$ENV_FILE"
+# Encryption keys
+SECRET_KEY_BASE=$(openssl rand -hex 64)
+ENCRYPTION_DETERMINISTIC_KEY=$(openssl rand -base64 32)
+ENCRYPTION_PRIMARY_KEY=$(openssl rand -base64 32)
+ENCRYPTION_KEY_DERIVATION_SALT=$(openssl rand -base64 32)
+KEYGEN_ACCOUNT_ID=$(uuidgen)
+EOF
+  echo "Encryption keys appended to .env file"
+fi
+
+# Update or add admin credentials if not already set
+update_env "KEYGEN_ADMIN_EMAIL" "${KEYGEN_ADMIN_EMAIL:-me@email.com}"
+update_env "KEYGEN_ADMIN_PASSWORD" "${KEYGEN_ADMIN_PASSWORD:-123456}"
+
+echo ".env file updated with necessary keys and values"
+
+# Run postgres and redis services
+docker-compose up -d postgres redis
+
+# Wait for PostgreSQL to be ready
+echo "Waiting for PostgreSQL to be ready..."
+until docker-compose exec -T postgres pg_isready; do
+  echo "PostgreSQL is unavailable - sleeping"
+  sleep 1
+done
+echo "PostgreSQL is up and running"
+
+# Run setup service
+echo "Running setup service..."
+docker-compose run --rm setup
```

Comment:
Is there a way to make it so that the setup doesn't run every time the script is run? The `setup` script runs `rails db:schema:load`, which wipes the database, i.e. a dangerous action for a "turn-key" setup like Compose.

The `setup` command is only meant to be run once during initial install, to create the account and admin user. This is why in https://github.com/keygen-sh/keygen-api/issues/727#issuecomment-1590322284, I suggested maybe we skip the `setup` command entirely for Compose, and use seeds instead,  which are safe to run multiple times (and recommended to run on each deploy to apply e.g. new permissions and event types).

Hope that makes sense? wdyt?